src/de.rs

```diff
@@ -103,3 +103,16 @@ impl<'de> Deserialize<'de> for Box<Bytes> {
         Ok(bytes.into())
     }
 }
+
+impl<'de, T> Deserialize<'de> for Option<T>
+where
+    T: Deserialize<'de> + serde::de::Deserialize<'de> + From<&'de [u8]>,
```

Comment:
Ideally this would only require `T: Deserialize<'de>`, not the other two bounds.